packages/monitor-v2/src/llm-bot/common.ts

```diff
@@ -94,63 +209,56 @@ export abstract class OptimisticOracleClient<R extends OptimisticOracleRequest>
    * @returns A new instance of OptimisticOracleClient.
    */
   protected abstract createClientInstance(
-    requests: R[],
-    fetchedBlockRanges: [number, number][]
+    requests: Map<string, R>,
+    fetchedBlockRange: BlockRange
   ): OptimisticOracleClient<R>;
 
   /**
-   * Retrieves Optimistic Oracle requests within the specified block range.
-   * @param blockRange The block range to fetch requests from.
-   * @returns A Promise that resolves to an array of OptimisticOracleRequest objects.
+   * Returns a copy of the updated Requests by fetching new Oracle requests updates.
+   * @param blockRange The new blockRange to fetch requests from.
+   * @returns A Promise that resolves to a copy of the updated Requests map.
    */
-  protected abstract fetchOracleRequests(blockRange: [number, number]): Promise<R[]>;
+  protected abstract updateOracleRequests(blockRange: BlockRange): Promise<Map<string, R>>;
 
   /**
    * Updates the OptimisticOracleClient instance by fetching new Oracle requests within the specified block range. Returns a new instance.
    * @param blockRange (Optional) The block range to fetch new requests from.
    * @returns A Promise that resolves to a new OptimisticOracleClient instance with updated requests.
    */
-  async updateWithBlockRange(blockRange?: [number, number]): Promise<this> {
-    let range: [number, number];
+  async updateWithBlockRange(blockRange?: BlockRange): Promise<OptimisticOracleClient<R>> {
+    let range: BlockRange;
     if (blockRange) {
-      if (blockRange[0] > blockRange[1]) throw new Error("Invalid block range");
+      if (blockRange[0] > blockRange[1])
+        throw new Error("Start block number should be less than or equal to end block number");
       range = blockRange;
     } else {
       // Calculate the next block range to fetch
       const latestBlock = await this.provider.getBlockNumber();
-      const lastFetchedRange = this.fetchedBlockRanges[this.fetchedBlockRanges.length - 1];
-      const nextStartBlock = lastFetchedRange[1] + 1;
+      const nextStartBlock = this.fetchedBlockRange[1] + 1;
       if (nextStartBlock > latestBlock) return this; // no new blocks to fetch
       range = [nextStartBlock, latestBlock];
     }
     const [startBlock, endBlock] = range;
-    if (
-      this.fetchedBlockRanges.some(([s, e]) => (s <= startBlock && startBlock <= e) || (s <= endBlock && endBlock <= e))
-    )
-      throw new Error("Block range already fetched");
 
-    // Add new range to the list of fetched ranges and sort them by start block number
-    const newRanges = [...this.fetchedBlockRanges, range].sort(([s1], [s2]) => s1 - s2);
+    // Throw an error if the new range doesn't directly follow the last fetched range
+    const lastFetchedEndBlock = this.fetchedBlockRange[this.fetchedBlockRange.length - 1];
```

Comment:
would this be the same as this.fetchedBlockRange is of type BlockRange that has exactly 2 elements:
```suggestion
    const lastFetchedEndBlock = this.fetchedBlockRange[1];
```